### PR TITLE
Add support for non-prose elements

### LINKED ISSUE #82: Support for auto `<a>` generation for internal links.
As noted in PR https://github.com/SMPTE/ag-06/pull/12, support is needed for internal links. 

Example from this PR (line 160-161 of main.html):
```
<li><cite id="element-smpte-ag-06a">SMPTE AG-06a</cite>:2022-12-06, New Project Form
        <a>element/AG-06a-New-Project Form-2022-12-06.docx</a></li>
```

### PR DESCRIPTION
**Pull request by palemieux**
Closes #82

### CODE REVIEW

File: doc/main.html [hunk 1]
```diff
@@ -229,6 +231,10 @@ <h4>Overall structure</h4>
               <td><a>Annex section</a></td>
               <td>0 or more</td>
             </tr>
+            <tr>
+              <td><a>non-prose elements section</a></td>
```
Comment (SteveLLamb) [inline doc/main.html:235]: To be in line with AG-16:
```suggestion
              <td><a>Additional elements section</a></td>
```

File: smpte.js [hunk 2]
```diff
@@ -422,6 +422,55 @@ function insertBibliography(docMetadata) {
...
+  }
+
+  sec.classList.add("annex");
+
+  const intro = document.createElement("p");
+  intro.innerText = "This annex lists non-prose elements of this document."
+  sec.insertBefore(intro, sec.firstChild);
+
+  const h2 = document.createElement("h2");
+  h2.innerText = "Non-prose elements";
```
Comment (SteveLLamb) [inline smpte.js:445]: Still wrong header section name

```suggestion
  h2.innerText = "Additional elements";
```

File: doc/main.html [hunk 3]
```diff
@@ -482,6 +488,38 @@ <h4>Annex section(s)</h4>
...
+        <h4>Non-prose elements section</h4>
+
+        <p>The <dfn>non-prose elements section</dfn> collects the non-prose elements of the document.</p>
+
+        <p>The <code>id</code> attribute shall be present on the <code>section</code> element and equal to
+        <code>sec-elements</code>.</p>
+
+        <p>The heading of the section shall not be present.</p>
+
+        <p>If present, the section shall contain a single ordered list <code>ol</code> where each element <code>li</code>
```
[the remark this answers is not in the document]
Comment (palemieux) [reply doc/main.html:501]: Can you provide a sample? I cannot reproduce the error.

[the remark this answers is not in the document]
Comment (SteveLLamb) [reply doc/main.html:501]: Try adding this after `sec-elements`:
```
<section class="annex" id="sec-checking-annexes">
    <h2>Checking Annexes</h2>
    <p>Checking annexes</p>
 </section>
```
Works fine both with and without the annex class defined prior to `sec-elements`. After throws the error.

[the remark this answers is not in the document]
Comment (SteveLLamb) [reply doc/main.html:501]: Full error message:
```
TypeError: Cannot read properties of undefined (reading 'at')
    at validateBody (file:///Users/llambs/LLux%20Operon%20Dropbox/Steve%20LLamb/Documents/DCin/Repos/SMPTE/html-pub/js/validate.mjs:357:33)
    at smpteValidate (file:///Users/llambs/LLux%20Operon%20Dropbox/Steve%20LLamb/Documents/DCin/Repos/SMPTE/html-pub/js/validate.mjs:58:3)
    at main (file:///Users/llambs/LLux%20Operon%20Dropbox/Steve%20LLamb/Documents/DCin/Repos/SMPTE/html-pub/scripts/build.mjs:422:3)
    at file:///Users/llambs/LLux%20Operon%20Dropbox/Steve%20LLamb/Documents/DCin/Repos/SMPTE/html-pub/scripts/build.mjs:456:1
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)
```

[the remark this answers is not in the document]
Comment (palemieux) [reply doc/main.html:501]: Ok. Bibliography had to be present.

[the remark this answers is not in the document]
Comment (palemieux) [reply doc/main.html]: See https://github.com/SMPTE/html-pub/pull/83/commits/56d99fb3835b1361828ec470897963951193fb24

**Review by SteveLLamb — APPROVED** [review]

**Pull request merged**